From 166c3126ff542fe0413fa35e7b647b23ab4ccb45 Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Wed, 7 Jul 2021 15:52:10 +0300
Subject: [PATCH 29/29] s32: xen: Update u-boot env to use generated boot
 script

Updated the u-boot environment to load the u-boot script and
execute it, as part of the "bootcmd" variable, instead of
manually setting up the u-boot env required for booting a
Xen setup.

The u-boot script does not include the final booti command,
therefore, the booting process can be stopped and the script
can be manually fetched and run, without automatically also booting
the Xen image. This can be useful when a user needs to dynamically
amend the environment or device-tree before booting.

Upstream-Status: Pending

Issue: ALB-6618
Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 include/configs/s32.h | 32 +++++---------------------------
 1 file changed, 5 insertions(+), 27 deletions(-)

diff --git a/include/configs/s32.h b/include/configs/s32.h
index 2d53c75498..449633b989 100644
--- a/include/configs/s32.h
+++ b/include/configs/s32.h
@@ -372,36 +372,14 @@
 #define CONFIG_FLASHBOOT_RAMDISK " ${ramdisk_addr} "
 
 #ifdef CONFIG_XEN_SUPPORT
-#ifdef CONFIG_TARGET_TYPE_S32GEN1_SIMULATOR
-#define XEN_LOAD_FILES "setenv filesize a00000; "
-#else
-#define XEN_LOAD_FILES \
-	"fatload mmc 0 ${fdt_addr} ${fdt_file}; " \
-	"fatload mmc 0 ${loadaddr} xen; " \
-	"fatload mmc 0 ${dom0_addr} Image; "
-#endif
 #define XEN_EXTRA_ENV_SETTINGS \
-	"dom0_addr=0x90000000\0" \
-	"dom0_mem=192M\0" \
-	"dom0_bootargs=\"console=hvc0 root=/dev/mmcblk0p2 rootwait rw\"\0" \
-	"bootargs_setup=setenv bootargs dom0_mem=${dom0_mem} bootscrub=0 " \
-		"console=dtuart dtuart=serial0\0" \
-	"dom0_node_setup=fdt set /chosen/module@0 compatible " \
-		"\"xen,linux-zimage\" \"xen,multiboot-module\"; " \
-		"fdt set /chosen/module@0 reg <${dom0_addr} 0x${filesize} >; " \
-		"fdt set /chosen/module@0 bootargs ${dom0_bootargs} \0" \
-	"chosen_node_setup=fdt addr ${fdt_addr} 0x40000; fdt resize 1024; " \
-		"fdt chosen; fdt set /chosen \\\\\#address-cells <1>; " \
-		"fdt set /chosen \\\\\#size-cells <1>; " \
-		"fdt mknod /chosen module@0;\0" \
-	"clear_default_chosen=fdt rm /chosen stdout-path; " \
-		"fdt rm /chosen linux,initrd-start; " \
-		"fdt rm /chosen linux,initrd-end; \0" \
-	"updatexenfdt=run chosen_node_setup; run dom0_node_setup; " \
-		"run bootargs_setup;\0" \
+	"script_addr=0x80200000\0" \
+	"mmcpart_ext=" __stringify(MMC_PART_EXT) "\0" \
 
 #define XEN_BOOTCMD \
-	XEN_LOAD_FILES "run updatexenfdt; booti ${loadaddr} - ${fdt_addr}"
+	"ext4load mmc ${mmcdev}:${mmcpart_ext} ${script_addr} " \
+		"boot/${script}; source ${script_addr}; " \
+		"booti ${xen_addr} - ${fdt_addr};"
 #else
 #define XEN_EXTRA_ENV_SETTINGS  ""
 #endif
-- 
2.17.1

